Enterprise-CMCS/macpro-onemac
# Oy2 23373 - add approved effective date and final disposition date to detail view

Story: https://qmacbis.atlassian.net/browse/OY2-23373
Endpoint: See github-actions bot comment

### Details

Add seatool data to detail view for approved effective date and final disposition date.
If Approved effective date does not exist then use actual effective date else show -- --
Only show final disposition date if status is Approved, Disapproved, or Withdrawn; this date comes from the seatool status date.

### Changes

- Updated buildAnyPackage to translate and add the new fields into package builder
- added new fields to detail view as formatted dates


### Test Plan

1. Submit a package
2. manually enter a SEATOOL record with approved status and approved effective date
3. verify values appear on details page
4. repeat with actual effective date

sample mock seatool record is as follows but could probably be trimmed to just the relevant fields:
{
 "pk": "MD-23-2323",
 "sk": "SEATool#1686802418976",
 "ACTIONTYPES": [
  {
   "ACTION_ID": 84,
   "ACTION_NAME": "Amend",
   "PLAN_TYPE_ID": 124
  }
 ],
 "ACTION_OFFICERS": null,
 "CALLHELDREASONS": null,
 "CODEAFTERINITACCESS": [
  {
   "CODE_AFTER_INIT_ASSESS_DESC": "Same",
   "CODE_AFTER_INIT_ASSESS_ID": 3
  }
 ],
 "COMPONENTS": [
  {
   "ALERTS_INBOX_ADDRESS": null,
   "COMPONENT_ID": 73,
   "COMPONENT_NAME": "CAHPG"
  }
 ],
 "COMPONENTS_SP": [
  {
   "ALERTS_INBOX_ADDRESS": null,
   "COMPONENT_ID": 76,
   "COMPONENT_NAME": "Test 9"
  }
 ],
 "FM_ANALYST": null,
 "GSI1pk": "SEATool#Medicaid_SPA",
 "GSI1sk": "MD-23-2323",
 "LEAD_ANALYST": [
  {
   "EMAIL": "abc@exampl.com",
   "FIRST_NAME": "Demo1",
   "INITIALS": "A",
   "LAST_NAME": "Test1",
   "OFFICER_ID": 3996,
   "POSITION_ID": null,
   "TELEPHONE": null
  }
 ],
 "OCD_REVIEW": [
  {
   "OCD_REVIEW_DESCRIPTION": "No",
   "OCD_REVIEW_ID": 2
  },
  {
   "OCD_REVIEW_DESCRIPTION": "Yes",
   "OCD_REVIEW_ID": 1
  }
 ],
 "PLAN_TYPES": [
  {
   "PLAN_TYPE_ID": 124,
   "PLAN_TYPE_NAME": "Medicaid SPA"
  }
 ],
 "PRIORITY_CODES": [
  {
   "PRIORITY_CODE": "P1",
   "PRIORITY_CODE_DESCRIPTION": "Must go through OCD",
   "PRIORITY_CODE_ID": 1
  },
  {
   "PRIORITY_CODE": "P3",
   "PRIORITY_CODE_DESCRIPTION": "Routine action test",
   "PRIORITY_CODE_ID": 3
  }
 ],
 "PRIORITY_COMPLEXITY": null,
 "PROGRAM_ANALYST": null,
 "RAI": null,
 "REGION": [
  {
   "ALERTS_INBOX_ADDRESS": "SPA_Waivers_Philadelphia_R03@cms.hhs.gov",
   "REGION_ID": "3",
   "REGION_NAME": "Philadelphia"
  }
 ],
 "REVIEW_POSITION": null,
 "RO_ANALYST": [
  {
   "EMAIL": null,
   "FIRST_NAME": "Test1",
   "INITIALS": null,
   "LAST_NAME": "Test2",
   "OFFICER_ID": 3993,
   "POSITION_ID": null,
   "TELEPHONE": null
  }
 ],
 "SP1115": null,
 "SPA_TYPE": null,
 "SPW_STATUS": [
  {
   "SPW_STATUS_DESC": "Approved",
   "SPW_STATUS_ID": 4
  }
 ],
 "SP_APD": null,
 "SP_APD_SUB_TYPE": null,
 "SP_EARLY_ALERTS": {
  "ALERTFIELD": [
   {
    "ACTIVE": true,
    "EA_DROP_DOWN_VALUES": null,
    "EA_FIELD_ID": 11,
    "EA_LABEL": "Reduced Eligibility",
    "EA_TYPE_ID": 1,
    "IS_APD": false,
    "PRIORITY_CODE_ID": 1,
    "PRIORITY_OPERATOR": null,
    "PRIORITY_VALUE": "true"
   },
   {
    "ACTIVE": true,
    "EA_DROP_DOWN_VALUES": null,
    "EA_FIELD_ID": 24,
    "EA_LABEL": "Meets expedited criteria",
    "EA_TYPE_ID": 1,
    "IS_APD": null,
    "PRIORITY_CODE_ID": 3,
    "PRIORITY_OPERATOR": null,
    "PRIORITY_VALUE": "true"
   }
  ],
  "EARLYALERT": [
   {
    "EA_FIELD_ID": 11,
    "EMAIL_SENT": null,
    "ID_NUMBER": "MD-23-2323",
    "TEXT_DD_VAL": null,
    "YES_NO_VAL": false
   },
   {
    "EA_FIELD_ID": 24,
    "EMAIL_SENT": null,
    "ID_NUMBER": "MD-23-2323",
    "TEXT_DD_VAL": null,
    "YES_NO_VAL": true
   }
  ]
 },
 "SP_IMPACT_FUNDING": null,
 "SP_TYPE": null,
 "STATES": [
  {
   "PRIORITY_FLAG": false,
   "REGION_ID": "3",
   "STATE_CODE": "MD",
   "STATE_NAME": "Maryland"
  }
 ],
 "STATE_PLAN": {
  "ACTION_TYPE": 84,
  "ACTUAL_EFFECTIVE_DATE": null,
  "ADDED_COST": null,
  "ALERT_90_DAYS_DATE": 1681084800000,
  "ALERT_MILESTONE1_DAYS": 59,
  "ALERT_MILESTONE2_DAYS": null,
  "ALERT_MILESTONE3_DAYS": null,
  "ALERT_MILESTONE4_DAYS": null,
  "APPROVAL_DOCS_RECEIVED": false,
  "APPROVAL_STATUS_TYPE": "test",
  "APPROVED_EFFECTIVE_DATE": 1699903418974,
  "ATTACHED_SPA": null,
  "BACKUP_FM_ANALYST_ID": null,
  "BACKUP_PROGRAM_ANALYST_ID": null,
  "BLOCKING_SPAS_MEMO": null,
  "BUDGET_IMPACT": null,
  "BUDGET_IMPACT_VALUE": null,
  "BUDGET_NEUTRALITY_ESTABLISHED_FLAG": false,
  "BUDGET_NEUTRALITY_STATUS_MEMO": null,
  "CALL_HELD": true,
  "CALL_HELD_REASON_ID": null,
  "CHANGED_DATE": 1673293365150,
  "CODE_AFTER_INIT_ASSESS_ID": 3,
  "COMPANION_LETTER_RECEIVED_DATE": null,
  "COMPANION_LETTER_REQUESTED_DATE": null,
  "COMPONENT_ID": 73,
  "CO_SUBMISSION_DATE": 1673222400000,
  "CURRENT_WAIVER_EXPIRES_DATE": null,
  "CURRENT_WAIVER_TE": null,
  "DATE_OF_CODING_CHANGE": 1673222400000,
  "DATE_SENT_PSCCAS": null,
  "DAYS_EXTENSION_NUMBER": null,
  "ELIMINATED_COST": null,
  "END_DATE": null,
  "FISCAL_QUARTER": null,
  "FISCAL_YEAR": null,
  "FRT_DATE": 1673222400000,
  "GAP": null,
  "GAP2": null,
  "GAP2_NA": null,
  "GAP3": null,
  "GAP3_NA": null,
  "GAP_NA": null,
  "GUIDANCE_DOCS_SUBMITTED": null,
  "ID_NUMBER": "MD-23-2323",
  "INITIAL_SUBMISSION_COMPLETE": true,
  "LEAD_ANALYST_ID": 3996,
  "MISSING_INFORMATION": null,
  "MMDL_IMPORT": null,
  "OCD_REVIEW_COMMENTS_MEMO": "test",
  "OCD_REVIEW_ID": 1,
  "ORGANIZATION_CHANGE": null,
  "PENDING_CONCURRENCE_DATE": null,
  "PLAN_TYPE": 124,
  "PRIORITY_CODE_ID": 3,
  "PRIORITY_COMMENTS_MEMO": "test",
  "PRIORITY_COMPLEXITY_ID": null,
  "PROPOSED_DATE": 1678233600000,
  "PUBLICHEALTH_STATEEMERGENCY": null,
  "REGION_ID": "3",
  "REMARKS_MEMO": null,
  "REVIEW_POSITION_ID": null,
  "RO_ANALYST_ID": 3993,
  "SPA_TYPE_ID": null,
  "SPW_IMPORT": false,
  "SPW_STATUS_ID": 4,
  "START_CLOCK_DATE": 1673308800000,
  "STATE_CODE": "MD",
  "STATUS_DATE": 1686802418974,
  "STATUS_MEMO": null,
  "SUBMISSION_DATE": 1673222400000,
  "SUBMISSION_TYPE": null,
  "SUMMARY_MEMO": "approved status",
  "TEMPLATE_ISSUES": null,
  "TEMPLATE_ISSUES_MEMO": null,
  "TEMPLATE_ISSUES_RESOLVED": null,
  "TE_END_DATE": null,
  "TITLE_NAME": "OneMac Connection test",
  "TYPE_ID": null,
  "UPL_ACCEPTED": null,
  "UUID": "D2150892-0EEE-4413-A06B-1252303E334E"
 },
 "STATE_PLAN_SERVICETYPES": [
  {
   "ID_NUMBER": "MD-23-2323",
   "SERVICE_TYPE_ID": 61
  }
 ],
 "STATE_PLAN_SERVICE_SUBTYPES": [
  {
   "ID_NUMBER": "MD-23-2323",
   "SERVICE_SUBTYPE_ID": 340
  }
 ],
 "STOP_RESUME_DATES": 3
}
